tools/driver/autolink_extract_main.cpp

```diff
@@ -107,26 +107,38 @@ class AutolinkExtractInvocation {
 
 /// Look inside the object file 'ObjectFile' and append any linker flags found in
 /// its ".swift1_autolink_entries" section to 'LinkerFlags'.
-static void
+/// Return 'true' if there was an error, and 'false' otherwise.
+static bool
 extractLinkerFlagsFromObjectFile(const llvm::object::ObjectFile *ObjectFile,
-                                 std::vector<std::string> &LinkerFlags) {
+                                 std::vector<std::string> &LinkerFlags,
+                                 CompilerInstance &Instance) {
   // Search for the section we hold autolink entries in
   for (auto &Section : ObjectFile->sections()) {
     llvm::StringRef SectionName;
     Section.getName(SectionName);
     if (SectionName == ".swift1_autolink_entries") {
-      llvm::StringRef SectionData;
-      Section.getContents(SectionData);
+      llvm::Expected<llvm::StringRef> SectionData = Section.getContents();
+      if (!SectionData) {
+        std::string message;
+        {
```

Comment:
Why do you need the extra scope here?

Reply:
The raw_string_ostream flushes when it goes out of scope.

I'll remove the formatting/nfc changes

Reply:
Also, I'll make the changes in https://github.com/apple/swift/pull/24880